From 566136bb6a26ed6ef753434f31ef633de449cefc Mon Sep 17 00:00:00 2001
From: CrystalP <CrystalP@xbmc.org>
Date: Sun, 25 Dec 2011 21:34:27 -0500
Subject: [PATCH] fix playback of dvd, dvd iso and video_ts

m_SelectionStreams is empty when playing a DVD or iso image of a dvd.
m_SelectionStreams doesn't contain video streams when playing a video_ts
folder.
---
 xbmc/cores/dvdplayer/DVDPlayer.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDPlayer.cpp b/xbmc/cores/dvdplayer/DVDPlayer.cpp
index 4e3ea71..b868cd0 100644
--- a/xbmc/cores/dvdplayer/DVDPlayer.cpp
+++ b/xbmc/cores/dvdplayer/DVDPlayer.cpp
@@ -2255,6 +2255,8 @@ bool CDVDPlayer::IsPaused() const
 
 bool CDVDPlayer::HasVideo() const
 {
+  if (m_pInputStream && m_pInputStream->IsStreamType(DVDSTREAM_TYPE_DVD)) return true;
+
   return m_SelectionStreams.Count(STREAM_VIDEO) > 0 ? true : false;
 }
 
-- 
1.7.5.4

